deepchem/models/losses.py

```diff
@@ -465,6 +470,216 @@ def loss(inputs):
+            Local feature map from the encoder.
+        g_enc: torch.Tensor
+            Global features from the encoder.
+        index: torch.Tensor
+            Index of the graph that each node belongs to.
+        device: torch.device
+            Device to run the model on.
+        measure: str
+            The divergence measure to use for the unsupervised loss. Options are 'GAN', 'JSD',
+            'KL', 'RKL', 'X2', 'DV', 'H2', or 'W1'.
+        average_loss: bool
+            Whether to average the loss over the batch
+
+        Returns:
+        -------
+        loss: torch.Tensor
+            Local Global Loss value
+        """
+        num_graphs = g_enc.shape[0]
+        num_nodes = l_enc.shape[0]
+
+        pos_mask = torch.zeros((num_nodes, num_graphs)).to(self.device)
+        neg_mask = torch.ones((num_nodes, num_graphs)).to(self.device)
+        for nodeidx, graphidx in enumerate(index):
+            pos_mask[nodeidx][graphidx] = 1.
+            neg_mask[nodeidx][graphidx] = 0.
+
+        res = torch.mm(l_enc, g_enc.t())
+
+        E_pos = self.get_positive_expectation(res * pos_mask, self.measure,
+                                              self.average_loss)
+        E_pos = (E_pos * pos_mask).sum() / pos_mask.sum()
+        E_neg = self.get_negative_expectation(res * neg_mask, self.measure,
+                                              self.average_loss)
+        E_neg = (E_neg * neg_mask).sum() / neg_mask.sum()
+
+        return E_neg - E_pos
+
+    def global_global_loss(self, g_enc, g_enc1):
+        """
+        Compares the encodings of two molecular graphs and returns the loss between them based on the measure specified.
+        The encodings are generated by two separate encoders in order to maximize the mutual information between the two encodings.
+
+        Parameters:
+        ----------
+        g_enc: torch.Tensor
+            Global features from the encoder.
+        g_enc1: torch.Tensor
+            Global features from the separate encoder.
+        device: torch.device
+            Device to run the model on.
+        measure: str
+            The divergence measure to use for the unsupervised loss. Options are 'GAN', 'JSD',
+            'KL', 'RKL', 'X2', 'DV', 'H2', or 'W1'.
+        average_loss: bool
+            Whether to average the loss over the batch
+
+        Returns:
+        -------
+        loss: torch.Tensor
+            Global Global Loss value
+        """
+        num_graphs = g_enc.shape[0]
+
+        pos_mask = torch.eye(num_graphs).to(self.device)
+        neg_mask = 1 - pos_mask
+
+        res = torch.mm(g_enc, g_enc1.t())
+
+        E_pos = self.get_positive_expectation(res * pos_mask)
+        E_pos = (E_pos * pos_mask).sum() / pos_mask.sum()
+        E_neg = self.get_negative_expectation(res * neg_mask)
+        E_neg = (E_neg * neg_mask).sum() / neg_mask.sum()
+
+        return E_neg - E_pos
+
+    def get_positive_expectation(self, p_samples):
```

Comment:
This should be a utility function and not a method (instead of `self`, just accept `measure` as a direct argument)